packages/driver/src/cy/commands/misc.js

```diff
@@ -50,6 +58,13 @@ module.exports = (Commands, Cypress, cy) => {
           onRetry: resolveWrap,
         })
         .return(arg)
+        .timeout(options.timeout)
+        .catch(Promise.TimeoutError, () => {
+          $errUtils.throwErrByPath('wrap.timed_out', {
+            onFail: options._log,
+            args: { timeout: options.timeout },
+          })
+        })
```

Comment:
is this necessary? I believe cy.verifyUpcomingAssertions will await `arg` with the proper timeout, maybe that is causing the issue you describe

Reply:
It doesn't look like that is the case, ran a couple tests locally without these changes and they aren't respecting the timeout in the way intended with these changes (exhibits the same behavior as before we removed `all()`).

Aside from that and assuming that the inconsistent errors are fixed, it looks like what this change is doing is actually modifying the way that the `timeout` option had already been getting processed which may or may not be desired.